### DISCUSSION TITLE
Not pulling in Narrator from Amazon.co.uk

### DISCUSSION THREAD

**Discussion by Nuuki9** [seanap/Plex-Audiobook-Guide#50] [Q&A] [answered]
Hi,

I've been using this workflow for a while and overall its been working great - amazing work. One minor niggle I'm trying to close out is that its not pulling in the Narrator for me.

I'm using the amazon.co.uk script - I've enabled debug and when I check the output file the last script line its showing as running is 101, which is `while "bc-color-link" 99`. The next entries in the debug are a section covering "Total output". 

Interestingly there are tags set after 101 which are being processed such as Description (line 276). As such it doesn't seem to be the case that its actually stopping at line 101, but as the moment I'm a bit stuck.

Any ideas?

**Comment by seanap** [comment, marked as answer]
Hey, thanks for finding this, the .co.uk script is failing to scrape the narrator for me as well.  

To troubleshoot this issue I compared side by side the `.com` and `.co.uk` web page source for the same book and searched for "narratorLabel".
I found that both sections look identical, and knowing that the .com script works I commented out (#) the whole narrator section in the .co.uk script and copied and pasted the narrator section from the .com script.  This worked and the narrator was found.  
This means the IF statement was failing for some reason, because like you pointed out the script was still finding data after the narrator section which tells me the ELSE condition was erroneously triggering.  
In the .co.uk script I scrolled down to the next section "Grouping / Series" and saw that I had an `unspace` line before the IF statement, so I added that to the narrator section and everything works now.

![image](https://user-images.githubusercontent.com/17012946/149624956-023906f5-df2f-4ccc-90a7-842943bc2240.png)

I will update the .co.uk file in the guide so you can either redownload the script or just add the unspace line shown above on line 176.  Let me know if this fixes the issue for you.

**Reply by Nuuki9** [reply]
That worked perfectly - thank you so much for lookng into this so quickly and getting it sorted. Amazing work!
